okhttp-testing-support/src/main/java/okhttp3/PlatformRule.kt

```diff
@@ -144,22 +133,16 @@ open class PlatformRule @JvmOverloads constructor(
           System.err.println("Warning: ALPN Boot enabled unintentionally")
         }
       }
+
+      PlatformRegistry.resetForTests()
+
+      System.err.println("Running with Platform ${Platform.get()}")
     }
 
     @JvmStatic
     fun getPlatformSystemProperty(): String {
-      var property: String? = System.getProperty(PROPERTY_NAME)
-
-      if (property == null) {
-        property = when (Platform.get()) {
-          is ConscryptPlatform -> CONSCRYPT_PROPERTY
-          is Jdk8WithJettyBootPlatform -> CONSCRYPT_PROPERTY
```

Comment:
This was a bug

Reply:
Yikes

Reply:
In tests and likely never hit in practice